### PR TITLE
Throw an exception to spark when the data push fails and there are too many retries

### PR DESCRIPTION
**Pull request by Aiden-Dong**
## Proposed changes

Close related #issue (replace it with issue number if it exists).

Describe the overview of changes, and introduce why we need it.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by hf200012 — DISMISSED** [review]
LGTM

File: extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala [hunk 1]
```diff
@@ -81,14 +83,24 @@ private[sql] class DorisStreamLoadSink(sqlContext: SQLContext, settings: SparkSe
...
                     Thread.currentThread.interrupt()
                     throw new IOException("unable to flush; interrupted while doing another attempt", e)
                 }
             }
           }
+
+          if(!rowsBuffer.isEmpty){
+            logger.warn("Failed to load data on BE: {} node ", dorisStreamLoader.getLoadUrlStr)
+            var errMsg = String.format("Failed to load data on BE: %s node and exceeded the max retry times.", dorisStreamLoader.getLoadUrlStr)
+            throw new IOException(errMsg)
```
Comment (wangshuo128) [inline extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala:102]:
```suggestion
            throw new IOException(s"Failed to load data on BE: ${dorisStreamLoader.getLoadUrlStr} node and exceeded the max retry times.")
```

Comment (jackwener) [inline extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala]:
```suggestion
            throw new IOException(s"Failed to load data on BE: ${dorisStreamLoader.getLoadUrlStr} node and exceeded the max retry times.")
```

File: extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala [hunk 2]
```diff
@@ -81,14 +83,24 @@ private[sql] class DorisStreamLoadSink(sqlContext: SQLContext, settings: SparkSe
...
+                    logger.warn("Data that failed to load : " + dorisStreamLoader.listToString(rowsBuffer))
                     Thread.currentThread.interrupt()
                     throw new IOException("unable to flush; interrupted while doing another attempt", e)
                 }
             }
           }
+
+          if(!rowsBuffer.isEmpty){
+            logger.warn("Failed to load data on BE: {} node ", dorisStreamLoader.getLoadUrlStr)
+            var errMsg = String.format("Failed to load data on BE: %s node and exceeded the max retry times.", dorisStreamLoader.getLoadUrlStr)
```
Comment (jackwener) [inline extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala:101]:
```suggestion
```

**Review by hf200012 — DISMISSED** [review]
lgtm

File: extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala [hunk 3]
```diff
@@ -97,14 +100,24 @@ private[sql] class DorisSourceProvider extends DataSourceRegister
...
+                    logger.warn("Data that failed to load : " + dorisStreamLoader.listToString(rowsBuffer))
                     Thread.currentThread.interrupt()
                     throw new IOException("unable to flush; interrupted while doing another attempt", e)
                 }
             }
           }
+
+          if(!rowsBuffer.isEmpty){
+            logger.warn("Data that failed to load : " + dorisStreamLoader.listToString(rowsBuffer))
+            logger.warn("Failed to load data on BE: {} node ", dorisStreamLoader.getLoadUrlStr)
```
Comment (morningman) [inline extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala:118]: merge these 2 `logger.warn`

File: extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala [hunk 4]
```diff
@@ -81,14 +83,24 @@ private[sql] class DorisStreamLoadSink(sqlContext: SQLContext, settings: SparkSe
...
+                    logger.warn("Data that failed to load : " + dorisStreamLoader.listToString(rowsBuffer))
                     Thread.currentThread.interrupt()
                     throw new IOException("unable to flush; interrupted while doing another attempt", e)
                 }
             }
           }
+
+          if(!rowsBuffer.isEmpty){
+            logger.warn("Data that failed to load : " + dorisStreamLoader.listToString(rowsBuffer))
+            logger.warn("Failed to load data on BE: {} node ", dorisStreamLoader.getLoadUrlStr)
```
Comment (morningman) [inline extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala:101]: Merge these 2 `logger.warn`

Comment (Aiden-Dong) [reply extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisStreamLoadSink.scala:101]: ok

**Review by morningman — APPROVED** [review]

**Pull request merged**